src/pilot/assign_role.py

```diff
@@ -780,8 +784,9 @@ def configure_raid(ironic_client, node_uuid, role, os_volume_size_gb,
     target_raid_config = define_target_raid_config(
         role, drac_client)
 
-    if target_raid_config is None:
-        return False
+    if not target_raid_config or target_raid_config is None:
```

Comment:
This "if" statement is returning success when target_raid_config is None, which is converting a critical failure into success.  When a critical failure occurs, assign_role should abort, and the way to make that happen is when target_raid_config is None, return False.  Only when the target_raid_config dict contains a "logical_disk" key that maps to an empty list, should True be returned here.